File: 10070.java
import java.lang.*;
import java.util.*;
import java.math.*;

class Main{
    static BigInteger b400 = new BigInteger("400");
    static BigInteger b4 = new BigInteger("4");
    static BigInteger b100 = new BigInteger("100");
    static BigInteger b15 = new BigInteger("15");
    static BigInteger b55 = new BigInteger("55");

    static boolean isLeapYear(BigInteger y){
        return (
                (y.mod(b100) != BigInteger.ZERO) && 
                (y.mod(b4)   == BigInteger.ZERO)) ||
                (y.mod(b400) == BigInteger.ZERO);
    }
    static boolean isHuluFest(BigInteger y){
        return (y.mod(b15) == BigInteger.ZERO);
    }
    static boolean isBuluFest(BigInteger y){
        return isLeapYear(y) && (y.mod(b55) == BigInteger.ZERO);
    }
    public static void main(String args[]){
        int tc;
        Scanner sc = new Scanner(System.in);
        BigInteger y;
        boolean first = true;
        while(sc.hasNextBigInteger()){
            if(first) first = false;
            else    System.out.println();

            boolean ordYear = true;
            y = sc.nextBigInteger();
            if(isLeapYear(y)){
                ordYear = false;
                System.out.println("This is leap year.");
            }
            if(isHuluFest(y)){
                ordYear = false;
                System.out.println("This is huluculu festival year.");
            }
            if(isBuluFest(y)){
                ordYear = false;
                System.out.println("This is bulukulu festival year.");
            }
            if(ordYear){
                ordYear = false;
                System.out.println("This is an ordinary year.");
            }
        }
    }
}

File: 10070.cpp
#include <iostream>
#include <sstream>
#include <vector>
#include <string>
#include <cstdio>
#include <cstring>
#include <algorithm>
#include <queue>
#include <set>
#include <stack>
#include <map>
#include <list>
#include <cmath>

using namespace std;

typedef pair<int,int>   ii;
typedef vector<int>     vi;
typedef vector<ii>      vii;
typedef vector<vi>     vvi;
typedef vector<vii>    vvii;
typedef map<int,int>    mii;

bool isLeapYear(long long y){
    return (y % 100 != 0 && y % 4 == 0) || (y % 400 == 0);
}
bool isHuluFest(long long y){
    return (y % 15 == 0);
}
bool isBuluFest(long long y){
    return isLeapYear(y) && (y % 55 == 0);
}

int main(){
    bool first = true;
    long long y;
    while(cin >> y){
        if(first)   first = false;
        else    putchar('\n');

        bool ordYear = true;
        if(isLeapYear(y)){
            ordYear = false;
            cout << "This is leap year." << endl;
        }
        if(isHuluFest(y)){
            ordYear = false;
            cout << "This is huluculu festival year." << endl;
        }
        if(isBuluFest(y)){
            ordYear = false;
            cout << "This is bulukulu festival year." << endl;
        }
        if(ordYear){
            cout << "This is an ordinary year." << endl;
        }
    }

    return 0;
}
